packages/wrangler/src/dev.tsx

```diff
@@ -949,11 +950,20 @@ export function getBindings(
 		vectorize: configParam.vectorize,
 		constellation: configParam.constellation,
 		hyperdrive: configParam.hyperdrive.map((hyperdrive) => {
-			if (!hyperdrive.localConnectionString) {
+			const connectionStringFromEnv = getHyperdriveLocalConnectionStringFromEnv()
+			if (!connectionStringFromEnv || !hyperdrive.localConnectionString) {
 				throw new UserError(
-					`In development, you should use a local postgres connection string to emulate hyperdrive functionality. Please setup postgres locally and set the value of "${hyperdrive.binding}"'s "localConnectionString" to the postgres connection string in your wrangler.toml`
+					`When developing locally, you should use a local Postgres connection string to emulate Hyperdrive functionality. Please setup Postgres locally and set the value of the 'HYPERDRIVE_LOCAL_CONNECTION_STRING' variable or "${hyperdrive.binding}"'s "localConnectionString" to the Postgres connection string.`
 				);
 			}
+
+			// If there is a non-empty connection string specified in the environment,
+			// use that as our local connection stirng configuration.
+			if (connectionStringFromEnv) {
```

Comment:
```suggestion
			if (connectionStringFromEnv !== undefined && connectionStringFromEnv !== '') {
```
Prefer strict checks here.